plugins/jobtype/src/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -0,0 +1,409 @@
+          throws FileNotFoundException {
+    // sanity check
+
+    if (!folder.exists()) {
+      throw new FileNotFoundException();
+
+    }
+    if (!folder.isDirectory()) {
+      throw new IllegalStateException(
+              "execution jar is suppose to be in this folder, but the object present is not a directory: "
+                      + folder);
+    }
+
+    File[] matchingFiles = folder.listFiles(new FilenameFilter() {
+      @Override
+      public boolean accept(File dir, String name) {
+        if (name.matches(regex))
+          return true;
+        else
+          return false;
+      }
+    });
+
+    if (matchingFiles == null) {
+      throw new IllegalStateException(
+              "the ls command in ResolveExecutionJarName threw an IOException");
+    }
+
+    return matchingFiles;
+  }
+
+  /**
+   * Pass in a log file, this method will find all the hadoop jobs it has launched, and kills it
+   * 
+   * Only works with Hadoop2
+   * 
+   * @param logFilePath
+   * @param log
+   */
+  public static void killAllSpawnedHadoopJobs(String logFilePath, Logger log) {
+    Set<String> allSpawnedJobs = findApplicationIdFromLog(logFilePath, log);
+    log.info("applicationIds to kill: " + allSpawnedJobs);
+  
+    for (String appId : allSpawnedJobs) {
+      try {
+        killJobOnCluster(appId, log);
+      } catch (Throwable t) {
+        log.warn("something happened while trying to kill this job: " + appId, t);
+      }
+    }
+  }
+
+  /**
+   * <pre>
+   * Takes in a log file, will grep every line to look for the application_id pattern.
+   * If it finds multiple, it will return all of them, de-duped (this is possible in the case of pig jobs)
+   * This can be used in conjunction with the @killJobOnCluster method in this file.
+   * </pre>
+   * 
+   * @param logFilePath
+   * @return a Set. May be empty, but will never be null
+   */
+  public static Set<String> findApplicationIdFromLog(String logFilePath, Logger log) {
+  
+    File logFile = new File(logFilePath);
+  
+    if (!logFile.exists()) {
+      throw new IllegalArgumentException("the logFilePath does not exist: " + logFilePath);
+    }
+    if (!logFile.isFile()) {
+      throw new IllegalArgumentException("the logFilePath specified  is not a valid file: "
+              + logFilePath);
+    }
+    if (!logFile.canRead()) {
+      throw new IllegalArgumentException("unable to read the logFilePath specified: " + logFilePath);
+    }
+  
+    BufferedReader br = null;
+    Set<String> applicationIds = new HashSet<String>();
+    Pattern p = Pattern.compile(".* (application_\\d+_\\d+).*");
+  
+    try {
+      br = new BufferedReader(new FileReader(logFile));
+      String input;
+  
+      // finds all the application IDs
+      while ((input = br.readLine()) != null) {
+        Matcher m = p.matcher(input);
+        if (m.find()) {
+          String appId = m.group(1);
+          if (!applicationIds.contains(appId)) {
```

Comment:
Since applicationIds is a set, not sure it is needed to check before adding to set.